administrator/components/com_associations/models/fields/itemlanguage.php

```diff
@@ -40,47 +40,74 @@ protected function getOptions()
 
-		$referenceId         = $input->get('id', '');
-		$associatedComponent = $input->get('acomponent', '');
-		$associatedView      = $input->get('aview', '');
-		$extension           = $input->get('extension', '');
-		$realView            = $extension !== '' ? $extension : $associatedView;
+		$component     = AssociationsHelper::getComponentProperties($input->get('component', '', 'string'));
+		$referenceId   = $input->get('id', 0, 'int');
+		$realView            = !is_null($component->extension) ? $component->extension : $component->item;
 
-		$key = $extension !== '' ? 'com_categories.category|' . $extension : $associatedComponent . '.' . $associatedView;
-		$cp  = AssociationsHelper::getComponentProperties($key);
-		$associations      = call_user_func(array($cp->associations->gethelper->class, $cp->associations->gethelper->method), $referenceId, $realView);
+		JLoader::register($component->associations->gethelper->class, $component->associations->gethelper->file);
+
+		$associations      = call_user_func(
+				array(
+					$component->associations->gethelper->class, 
+					$component->associations->gethelper->method), 
+					$referenceId, $realView
+			);
 
 		// Get reference language.
-		$table = clone $cp->table;
+		$table         = clone $component->table;
 		$table->load($referenceId);
-
-		$referenceLanguage = $table->{$cp->fields->language};
+		$referenceLang = $table->{$component->fields->language};
+		$user          = JFactory::getUser();
 
 		$existingLanguages = JHtml::_('contentlanguage.existing', false, true);
 
 		foreach ($existingLanguages as $key => $lang)
 		{
-			if ($lang->value == $referenceLanguage)
+			// If is equal to reference language
+			if ($lang->value == $referenceLang)
 			{
 				unset($existingLanguages[$key]);
 			}
 			if (isset($associations[$lang->value]))
 			{
-				if ($associatedComponent != 'com_menus')
+				if ($component->component != 'com_menus')
 				{
 					parse_str($associations[$lang->value], $contents);
 					$removeExtra  = explode(":", $contents['id']);
-					$lang->value  = $lang->value . "|" . $removeExtra[0];
+					$itemId       = $removeExtra[0];
+					$lang->value  = $lang->value . "|" . $itemId;
 				}
 				else
 				{
+					$itemId      = $associations[$lang->value];
 					$lang->value = $lang->value . "|" . $associations[$lang->value];
 				}
+
+				$canEdit    = $user->authorise('core.edit', $component->assetKey . '.' . $itemId);
+				$table->load($itemId);
+
+				if (!is_null($table->{$component->fields->created_by}))
+				{
+					$canEditOwn = $user->authorise('core.edit.own', $component->assetKey . '.' . $itemId) && $table->{$component->fields->created_by} == $user->id;
+					$canEdit    = $canEdit || $canEditOwn;
+				}
+
+				if (!$canEdit)
+				{
+					$lang->disable = true;
+				}	
 			}
 			else
 			{
 				$lang->value .= '|0';
+				$canCreate    = $user->authorise('core.create', $component->assetKey);
+
+				if (!$canCreate)
+				{
+					$lang->disable = true;
+				}
```

Comment:
Actually whatt i just said is not correct...

because a category inside the componet can also have core.create persmission.

so we need to check this better